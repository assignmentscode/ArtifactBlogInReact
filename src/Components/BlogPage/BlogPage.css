body, html {
    height: 100%;
    width: 100%;
    font-weight: 10;
    margin: 0px;
  }
  .content-width {
    width: 70%;
    margin: auto;
  }
  .div-blogs {
    display: flex;
    justify-content: center;
    flex-wrap: wrap; 
    align-content: space-between;
    flex: 1 0 calc(33.333% - 20px);
  }
  .create-btn {
    display: flex;
    justify-items: center;
  }
  .btn-class {
    color: white;
    display: flex;
    justify-content: center;
    margin-bottom: 10px;
}
.btn {
    height: 50px;
    width: 150px;
    font-size: 1.2em;
    color: white;
    background-color: orange;
}