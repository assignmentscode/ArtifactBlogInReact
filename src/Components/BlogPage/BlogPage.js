import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Header from './../Header/Header';
import Footer from './../Footer/Footer';
import Card from './../Card/Card';
import './BlogPage.css';
const axios = require('axios');

class BlogPage extends Component {
    state = {
        Blogdata : [],
      }
    storeData = (responseObject) => this.setState({Blogdata: responseObject});
    componentDidMount () {
        return axios.get('https://api.myjson.com/bins/hc5ye')
        .then((result)=>this.storeData(result.data));
    }
    render(){
        const cards = this.state.Blogdata.map((cardData) => {
            return (<Card data={cardData} key={cardData.title}/>);
        });
        return(
            <div>
                <Header/>
                <div className="div-blogs content-width">
                    {cards}
                </div>
                <div className="btn-class">
                    <Link to ="/posts">
                        <button className="btn">Post New Blog</button>
                    </Link>
                </div>
                <Footer/>
            </div>
        );
  }
}

export default BlogPage;