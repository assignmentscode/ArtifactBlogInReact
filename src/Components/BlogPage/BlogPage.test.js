import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router} from 'react-router-dom';
import {shallow} from 'enzyme';
import BlogPage from './BlogPage';
const BlogData = require('./../../mockData/index.json');

describe('BlogPage', ()=> {
    it('should render a complete webPage', ()=>{
        const tree = renderer.create(<Router><BlogPage Blogdata={BlogData}/></Router>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    const wrapper = shallow(<Router><BlogPage Blogdata={BlogData}/></Router>);
    it('should have more than 2 items in div', ()=> {
        expect(wrapper.find('div').getElements()[0].props.children.length > 2).toEqual(true);
    });
    it('should have another div tag as child of div tag', ()=> {
        //console.log(wrapper.find('Card').getElements());
        expect(wrapper.find('Card').getElements().length > 0).toEqual(true);
    });
    it('should have 3 div elements', () => {
        //console.log(wrapper.find('div').getElements());
        expect(wrapper.find('div').getElements().length>1).toEqual(true);
    });
    it('should have the initial state empty', () =>{
        const wrapper = shallow(<Router><BlogPage/></Router>);
        expect(wrapper.instance().state.Blogdata).toEqual([]);
      });
      it('should now modify the initial state', async ()=> {
        await wrapper.instance().componentDidMount();
        expect(wrapper.instance().state.Blogdata.length>0).toEqual(true);
      });
});