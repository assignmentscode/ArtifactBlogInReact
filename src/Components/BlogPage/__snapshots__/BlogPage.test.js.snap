// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPage should render a complete webPage 1`] = `
<div>
  <header>
    <nav
      className="content-width"
    >
      <p>
        The Artifact
      </p>
      <p
        className="heading-content"
      >
        Culture & Art blog
      </p>
      <ul>
        <li>
          Blog
        </li>
        <li>
          About
        </li>
        <li>
          Contact
        </li>
      </ul>
    </nav>
  </header>
  <div
    className="div-blogs content-width"
  />
  <div
    className="btn-class"
  >
    <a
      href="/posts"
      onClick={[Function]}
    >
      <button
        className="btn"
      >
        Post New Blog
      </button>
    </a>
  </div>
  <footer
    className="footer"
  >
    <p
      className="content-width"
    >
      Â©artifact.com 2019
    </p>
  </footer>
</div>
`;
